src/test/ui/proc-macro/issue-50493.stderr

```diff
@@ -1,26 +1,9 @@
 error[E0742]: visibilities can only be restricted to ancestor modules
-  --> $DIR/issue-50493.rs:9:12
+  --> $DIR/issue-50493.rs:8:12
    |
 LL |     pub(in restricted) field: usize,
    |            ^^^^^^^^^^
 
-error[E0616]: field `field` of struct `Restricted` is private
-  --> $DIR/issue-50493.rs:6:10
-   |
-LL | #[derive(Derive)]
-   |          ^^^^^^ private field
-   |
-   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
-
-error[E0616]: field `field` of struct `Restricted` is private
-  --> $DIR/issue-50493.rs:6:10
-   |
-LL | #[derive(Derive)]
-   |          ^^^^^^ private field
-   |
-   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment:
Is this change in the diagnostic intentional?

Reply:
Visibility with incorrect path `restricted` is now recovered as public, which prevents the follow-up errors.